lib/TokenManager.js

```diff
@@ -32,7 +36,34 @@ function remove(storage, key) {
   storage.setTokenStorage(tokenStorage);
 }
 
-function TokenManager(options) {
+function refresh(sdk, storage, key) {
+  try {
+    var token = get(storage, key);
+    if (!token) {
+      throw new AuthSdkError('The tokenManager has no token for the key: ' + key);
+    }
+  } catch (e) {
+    return Q.reject(e);
+  }
+
+  var responseType;
+  if (token.accessToken) {
+    responseType = 'token';
+  } else {
+    responseType = 'id_token';
+  }
+
+  return sdk.token.getWithoutPrompt({
+    responseType: responseType,
+    scope: token.scopes
+  })
+  .then(function(freshToken) {
+    add(storage, key, freshToken);
+    return freshToken;
+  });
```

Comment:
I think it makes sense to remove the existing token; I'll add that.

Listening to events will be in a follow-up PR, probably bundled with autoRefresh.